docs/downloader.rst

```diff
@@ -0,0 +1,169 @@
+For example, deciding "I like the other device's file better" would mean moving the ``.conflict`` file over top of the existing one.
+Deciding "I like mine better" means simply deleting the ``.conflict`` file.
+A more-complex strategy of merging the contents would mean updating the existing file **before** deleting the ``.conflict`` file.
+
+(Meejah believes the above accurately describes what Tahoe 1.14.0 magic-folder does).
+
+This doesn't mean it's the best "API" for conflict resolution (nor does it need to remain the only one).
+In fact, it likely is not a good API for any but motivated, advanced users and also seems like a bad API for other programs.
+
+(Not implemented). In keeping with other new development in magic-folder, there is an explicit HTTP API to resolve a conflict.
+For now, we limit this to selecting "mine" or "theirs".
+A future extension might wish to provide a way to provide completely new content (e.g. if the user edited a diff, for example).
+
+
+``GET /v1/conflicts/<folder-name>``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+(Not implemented).
+
+Returns a list (possibly empty) of local filesystem paths corresponding to each Snapshot that is currently in a Conflict state in the given magic-folder.
+
+Our content is in the path itself.
+The conflicting "other" content is in ``<path>.conflict-<name>`` where ``<name>`` is the petname of the user who is provided the conflicted content.
+
+Justification: we need somewhere for "theirs" versis "my" content .. I think we should still reflect this on the filesystem, even if the *API to manipulate it* is no longer there.
+This makes it more obvious for CLI users that they should check the conflicts list; the only alternative would seem to be "run some command occasionally to check for conflicts".
+I will forget to run this command.
+
+
+``POST /v1/resolve_conflict/<folder-name>?path=<some-path>&resolution=<theirs|mine>``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+(Not implemented).
+
+The ``path`` query argument is required.
+It must be a filesystem path relative to the selected magic-folder.
+
+The ``resolution`` query argument is required.
+It must be either the string ``theirs`` or the string ``mine``.
+
+It is an error if the given ``path`` in the given magic-folder is not currently in a conflicted state. In this case the response code is **404 Not Found** (XXX is this appropriate?)
+
+If the resolution is ``theirs`` then the file at ``<path>.theirs.<name>`` is moved to ``<path>`` and a new (local) Snapshot is created (with two parents).
+
+If instead the resolution is ``mine`` then the file at ``<path>.theirs.<name>`` is deleted and a new (local) Snapshot is created (with two parents).
+
+The response is delayed until the local state tracking the new Snapshot has been created.
+
+The response code is **CREATED** and the **Content-Type** is ``application/json``.
+
+The response body follows the form of this example::
+
+  {}
```

Comment:
Not reviewed, since this isn't going to be part of v1. I wonder if there is value in including these docs, if they are not yet implemented.

Reply:
moved conflict stuff to proposal doc